### PR TITLE
Adding fluid dynamics to ROM App

### PR DESCRIPTION
**Pull request by Rbravo555**
This PR generalizes the ROM structure to cope with fluid dynamics using the monolithic solver

### CODE REVIEW

**Review by philbucher — CHANGES_REQUESTED** [review]
there are a few huge files, please reduce the size

File: applications/RomApplication/python_scripts/navier_stokes_solver_vmsmonolithic_rom.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import KratosMultiphysics.FluidDynamicsApplication as KratosCFD
+
+# Import base class file
+from KratosMultiphysics.FluidDynamicsApplication.navier_stokes_solver_vmsmonolithic import NavierStokesSolverMonolithic
+import KratosMultiphysics.RomApplication as romapp
+
+def CreateSolver(model, custom_settings):
+    return ROMSolver(model, custom_settings)
+
+class ROMSolver(NavierStokesSolverMonolithic):
```
Comment (RiccardoRossi) [inline applications/RomApplication/python_scripts/navier_stokes_solver_vmsmonolithic_rom.py:14]: this name looks a little too generic ...

**Review by RiccardoRossi — COMMENTED** [review]
I agree with the comment of @philbucher we need the tests to be smaller than this, otherwise they will take too long to run and they will make our repo enormous...

**Comment by philbucher** [comment]
In order to not have the huge files in the repo you can do two things:
- Instead of doing a merge commit (this is what we normally do) do a squash commit. Then all changes are squashed into one commit
- reset the last commit locally and force push, this way it will be gone. After that add the test with smaller files

BTW I see that you are using non-standard python modules (e.g. numpy). For those you need to add `skiptest` otherwise those tests will fail when running them on a machine where those modules are not available
[This](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/CoSimulationApplication/tests/co_simulation_test_factory.py#L11-L15) is how we do this in the CoSimApp

File: applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+class FluidDynamicsAnalysisROM(FluidDynamicsAnalysis):
+
+    def __init__(self,model,project_parameters):
+        super(FluidDynamicsAnalysisROM,self).__init__(model,project_parameters)
+
+    #### Internal functions ####
+    def _CreateSolver(self):
+        """ Create the Solver (and create and import the ModelPart if it is not alread in the model) """
+        ## Solver construction
+        with open('RomParameters.json') as rom_parameters:
```
Comment (philbucher) [inline applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py:20]: why a separate json and not in ProjectParameters.json?

File: applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import KratosMultiphysics.RomApplication as romapp
+from KratosMultiphysics.RomApplication import python_solvers_wrapper_rom as solver_wrapper
+from KratosMultiphysics.FluidDynamicsApplication.fluid_dynamics_analysis import FluidDynamicsAnalysis
+
+import json
+
+class FluidDynamicsAnalysisROM(FluidDynamicsAnalysis):
+
+    def __init__(self,model,project_parameters):
+        super(FluidDynamicsAnalysisROM,self).__init__(model,project_parameters)
```
Comment (philbucher) [inline applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py:14]:
```suggestion
        super().__init__(model,project_parameters)
```

this is possible with python3 (we dropped python2 support :tada:)

File: applications/RomApplication/python_scripts/navier_stokes_solver_vmsmonolithic_rom.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    def __init__(self, model, custom_settings):
+        super(ROMSolver, self).__init__(model, custom_settings)
```
Comment (philbucher) [inline applications/RomApplication/python_scripts/navier_stokes_solver_vmsmonolithic_rom.py:17]:
```suggestion
        super().__init__(model, custom_settings)
```

File: applications/RomApplication/python_scripts/navier_stokes_solver_vmsmonolithic_rom.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    def GetDefaultSettings(cls):
+        default_settings = KratosMultiphysics.Parameters("""
+        {
+            "rom_settings": {
+            "nodal_unknowns": [ "TEMPERATURE" ],
+            "number_of_rom_dofs": 3
+            }
+        }
+        """)
+        default_settings.AddMissingParameters(super(ROMSolver,cls).GetDefaultSettings())
```
Comment (philbucher) [inline applications/RomApplication/python_scripts/navier_stokes_solver_vmsmonolithic_rom.py:31]:
```suggestion
        default_settings.AddMissingParameters(super().GetDefaultSettings())
```

File: applications/RomApplication/tests/fluid_dynamics_test_files/MainKratosROM.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+class TestFluidDynamicsROM(FluidDynamicsAnalysisROM):
+
+    def __init__(self,model,project_parameters,flush_frequency=10.0):
+        super(TestFluidDynamicsROM,self).__init__(model,project_parameters)
+        self.flush_frequency = flush_frequency
+        self.last_flush = time.time()
+        self.selected_time_step_solution_container = []
+
+    def FinalizeSolutionStep(self):
+        super(TestFluidDynamicsROM,self).FinalizeSolutionStep()
```
Comment (philbucher) [inline applications/RomApplication/tests/fluid_dynamics_test_files/MainKratosROM.py:20]:
```suggestion
        super().FinalizeSolutionStep()
```

File: applications/RomApplication/tests/fluid_dynamics_test_files/MainKratosROM.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+import sys
+import time
+import numpy as np
+import pdb
+
+class TestFluidDynamicsROM(FluidDynamicsAnalysisROM):
+
+    def __init__(self,model,project_parameters,flush_frequency=10.0):
+        super(TestFluidDynamicsROM,self).__init__(model,project_parameters)
```
Comment (philbucher) [inline applications/RomApplication/tests/fluid_dynamics_test_files/MainKratosROM.py:14]:
```suggestion
        super().__init__(model,project_parameters)
```

**Pull request closed without merging**